Up2dateService/Up2dateService/Installers/Choco/ChocoInstaller.cs

```diff
@@ -55,31 +68,43 @@ public bool IsPackageInstalled(Package package)
 
         public void Refresh()
         {
-            using (var ps = PowerShell.Create())
-            {
-                const string psCommand = @"choco list -li";
-                ps.AddScript(psCommand);
-                productCodes.Clear();
-                productCodes.AddRange(ps.Invoke<string>());
-            }
+            Process p = new Process();
+            p.StartInfo.FileName = "choco.exe";
+            p.StartInfo.Arguments = "list -l";
+            p.StartInfo.RedirectStandardOutput = true;
+            p.StartInfo.UseShellExecute = false;
+            p.Start();
+            p.WaitForExit();
+            productCodes.Clear();
+            productCodes.AddRange(p.StandardOutput.ReadToEnd()
+                .Split(new[] { Environment.NewLine },
+                    StringSplitOptions.RemoveEmptyEntries));
         }
 
         public void UpdatePackageInfo(ref Package package)
         {
             ChocoNugetInfo info = ChocoNugetInfo.GetInfo(package.Filepath);
             if (info == null || string.IsNullOrWhiteSpace(info.Id) || string.IsNullOrWhiteSpace(info.Version)) return;
-
             package.DisplayName = info.Title;
             package.Publisher = info.Publisher;
         }
 
         private static bool IsChocoInstalled()
         {
-            using (var ps = PowerShell.Create())
+            try
+            {
+                Process p = new Process();
+                p.StartInfo.FileName = "choco.exe";
+                p.StartInfo.Arguments = "--version";
+                p.StartInfo.RedirectStandardOutput = true;
+                p.StartInfo.UseShellExecute = false;
+                p.Start();
+                p.WaitForExit();
```

Comment:
I don't think we need to wait for exit.
If choco is not installed the p.Start() will throw exception immediately (please check it anyway).